### PR TITLE
Added option to sort lists by the translated title

### PR DESCRIPTION
**Pull request by bradymiller**
This is a modified solution for the following PR:
https://github.com/openemr/openemr/pull/229

### CODE REVIEW

**Comment by bradymiller** [comment]
@matrix-amiel 
What are your thoughts on this solution?

**Comment by bradymiller** [comment]
@sunsetsystems 
Since only required it in 1 spot, still think it requires a modularized solution (I'd prefer to see it used in a couple more places to get a practical idea of how to modularize it)?

**Comment by matrix-amiel** [comment]
I'ts look good solution.

File: library/options.inc.php [hunk 1]
```diff
@@ -122,7 +122,33 @@ function generate_select_list($tag_name, $list_id, $currvalue, $title, $empty_na
...
+        if ($GLOBALS['gb_how_sort_list'] == '0') {
+          //sort by seq
+          $lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ? AND activity=1 ORDER BY seq, title", array($list_id));
+        } else if ($GLOBALS['gb_how_sort_list'] == '1') {
+          // sort by title
+          if (($lang_id == '1' && !empty($GLOBALS['skip_english_translation'])) || !$GLOBALS['translate_lists']) {
+            $lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ? AND activity=1 ORDER BY title, seq", array($list_id));
+          } else {
+            $lres = sqlStatement("SELECT lo.option_id, " .
+              "IF(LENGTH(ld.definition),ld.definition,lo.title) AS title, " .
```
Comment (matrix-amiel) [inline library/options.inc.php]: Hi brady.
We must add to this query = "SELECT lo.option_id**, lo.is_default**......" the our second fix of the default option doesn't work without this column..

File: library/options.inc.php [hunk 2]
```diff
@@ -122,7 +122,33 @@ function generate_select_list($tag_name, $list_id, $currvalue, $title, $empty_na
...
+        // List order depends on language translation options.
+        //  (Note we do not need to worry about the list order in the algorithm
+        //   after the below code block since that is where searches for exceptions
+        //   are done which include inactive items or items from a backup
+        //   list; note these will always be shown at the bottom of the list no matter the
+        //   chosen order.)
+        $lang_id = empty($_SESSION['language_choice']) ? '1' : $_SESSION['language_choice'];
+        if ($GLOBALS['gb_how_sort_list'] == '0') {
+          //sort by seq
+          $lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ? AND activity=1 ORDER BY seq, title", array($list_id));
```
Comment (sunsetsystems) [inline library/options.inc.php]: The secondary sort is by title so you should use the logic below instead, only with ORDER BY being different.  It's not unusual to have many items with the same sequence number as a way to effectively sort by title.

**Comment by sunsetsystems** [comment]
For what it's worth the same logic appears in interface/super/edit_list.php.  Also there are versions for lang_languages instead of list_options in:

interface/login/login.php
interface/language/lang_definition.php
patients/index.php

File: library/options.inc.php [hunk 3]
```diff
@@ -122,7 +122,33 @@ function generate_select_list($tag_name, $list_id, $currvalue, $title, $empty_na
...
+        } else if ($GLOBALS['gb_how_sort_list'] == '1') {
```
Comment (bradymiller) [inline library/options.inc.php]: good point. will change to:
``````
} else { //$GLOBALS['gb_how_sort_list'] == '1'

File: library/options.inc.php (hunk 2 above)
Comment (bradymiller) [inline library/options.inc.php]: good point

File: library/options.inc.php (hunk 1 above)
Comment (bradymiller) [inline library/options.inc.php]: will do

File: library/options.inc.php [hunk 4]
```diff
@@ -136,7 +176,12 @@ function generate_select_list($tag_name, $list_id, $currvalue, $title, $empty_na
 			$got_selected = TRUE;
 		}
 		
-		$optionLabel = text(xl_list_label($lrow ['title']));
+                if ( !empty($lrow['already_translated']) && ($lrow['already_translated'] == 1) ) {
+		        $optionLabel = text($lrow ['title']);
+                }
+                else {
+                        $optionLabel = text(xl_list_label($lrow ['title']));
+                }
```
Comment (bradymiller) [inline library/options.inc.php]: this code may not be needed anymore

**Comment by bradymiller** [comment]
This is testing well. Thanks for all the input and gonna bring this into the codebase. Decided to take on the modularization of this logic on another day :)

**Comment by bradymiller** [comment]
Just committed this into the codebase. Thank you Amiel for the contribution.

**Pull request closed without merging**